script.js

```diff
@@ -0,0 +1,964 @@
+// Recipe Library - Spoonacular API with localStorage caching
+// backupData is loaded globally from backupdata.js script tag
+
+// === UI HELPERS ===
+
+/** Update sort button styling based on selection */
+function updateSortSelectedClass() {
+    const sortOptions = document.querySelectorAll('.filter-group.sort-group .filter-option');
+    sortOptions.forEach(option => {
+        const input = option.querySelector('input[type="radio"]');
+        if (input && input.checked) {
+            option.classList.add('selected');
+        } else {
+            option.classList.remove('selected');
+        }
+    });
+}
+
+// === API CONFIGURATION & GLOBAL STATE ===
+
+const API_CONFIG = {
+    BASE_URL: 'https://api.spoonacular.com',
+    API_KEY: '5607f37e52ff4b879d8f5006edbca556',  
+    ENDPOINTS: {
+        RANDOM: '/recipes/random'
+    }
+};
+
+// Global recipe storage
+let allRecipes = [];
+let recipes = [];
+let displayedRecipes = [];
+
+// Pagination settings
+let recipesPerPage = 15;
+let currentPage = 1;
+
+// === DOM ELEMENT REFERENCES ===
+
+const clearFiltersBtn = document.getElementById('clear-filters-btn');
+const randomRecipeBtn = document.getElementById('random-recipe-btn');
+const favoritesBtn = document.getElementById('favorites-btn');
+const recipeContainer = document.getElementById('recipe-container');
+const messageDisplay = document.getElementById('message-display');
+const searchInput = document.getElementById('search-input');
+
+
+// === API FUNCTIONS ===
+
+/**
+ * Merge cached and backup recipes, removing duplicates
+ * Pure function - no side effects or UI updates
+ * @returns {Array} - Combined recipes from cache and backup
+ */
+function getMergedRecipes() {
+    const cachedRecipes = getCachedRecipes() || [];
+    const mappedBackup = backupData.recipes.map(mapSpoonacularToLocal);
+    
+    // Merge both sources, removing duplicates by ID
+    const combinedRecipes = [...cachedRecipes];
+    const existingIds = new Set(cachedRecipes.map(r => r.id));
+    
+    mappedBackup.forEach(recipe => {
+        if (!existingIds.has(recipe.id)) {
+            combinedRecipes.push(recipe);
+        }
+    });
+    
+    return combinedRecipes;
+}
```

Comment:
I get what you're doing here, and nice job figuring that out. However, is it necessary?